docs/sphinx_extensions/dfhack/tool_docs.py

```diff
@@ -0,0 +1,175 @@
+                (command, bind.split('-'), context)]
+
+def scan_all_keybinds(root_dir):
+    """Get the implemented keybinds, and return a dict of
+    {tool: [(full_command, keybinding, context), ...]}.
+    """
+    keybindings = dict()
+    for root, _, files in os.walk(root_dir):
+        scan_keybinds(root, files, keybindings)
+    return keybindings
+
+
+def render_dfhack_keybind(command) -> List[nodes.paragraph]:
+    _KEYBINDS_RENDERED.add(command)
+    out = []
+    if command not in _KEYBINDS:
+        return out
+    for keycmd, key, ctx in _KEYBINDS[command]:
+        n = nodes.paragraph()
+        n += nodes.strong('Keybinding:', 'Keybinding:')
+        n += nodes.inline(' ', ' ')
+        for k in key:
+            n += nodes.inline(k, k, classes=['kbd'])
+        if keycmd != command:
+            n += nodes.inline(' -> ', ' -> ')
+            n += nodes.literal(keycmd, keycmd, classes=['guilabel'])
+        if ctx:
+            n += nodes.inline(' in ', ' in ')
+            n += nodes.literal(ctx, ctx)
+        out.append(n)
+    return out
+
+
+def check_missing_keybinds():
+    # FIXME: _KEYBINDS_RENDERED is empty in the parent process under parallel builds
+    # consider moving to a sphinx Domain to solve this properly
+    for missing_command in sorted(set(_KEYBINDS.keys()) - _KEYBINDS_RENDERED):
+        logger.warning('Undocumented keybindings for command: %s', missing_command)
+
+
+# pylint:disable=unused-argument,dangerous-default-value,too-many-arguments
+def dfhack_keybind_role(role, rawtext, text, lineno, inliner,
+                        options={}, content=[]):
+    """Custom role parser for DFHack default keybinds."""
+    return render_dfhack_keybind(text), []
+
+
+class DFHackToolDirectiveBase(sphinx.directives.ObjectDescription):
+    has_content = False
+    required_arguments = 0
+    optional_arguments = 1
+
+    def get_name_or_docname(self):
+        if self.arguments:
+            return self.arguments[0]
+        else:
+            return self.env.docname.split('/')[-1]
+
+    @staticmethod
+    def make_labeled_paragraph(label, content, label_class=nodes.strong, content_class=nodes.inline) -> nodes.paragraph:
+        return nodes.paragraph('', '', *[
+            label_class('', '{}: '.format(label)),
+            content_class('', content),
+        ])
+
+    @staticmethod
+    def wrap_box(*children: List[nodes.Node]) -> nodes.Admonition:
+        return nodes.admonition('', *children, classes=['dfhack-tool-summary'])
+
+    def render_content(self) -> List[nodes.Node]:
+        raise NotImplementedError
+
+    def run(self):
+        return [self.wrap_box(*self.render_content())]
+
+
+class DFHackToolDirective(DFHackToolDirectiveBase):
+    option_spec = {
+        'tags': dfhack.util.directive_arg_str_list,
+        'no-command': rst_directives.flag,
+    }
+
+    def render_content(self) -> List[nodes.Node]:
+        tag_nodes = [nodes.strong(text='Tags: ')]
```

Comment:
```suggestion
        tag_nodes = [nodes.strong(text='Tags:'), nodes.inline(text=' ')]
```

Reply:
otherwise the text rendering produces `**Tags: **tagname`, instead of `**Tags:** tagname`

Reply:
good catch!

Reply:
It came up previously when I was testing text rendering with the keybinding directive. I made the same fix there too.